### ISSUE TITLE
[BSP] - Incorrect `Range` in `Diagnostic`.

### ISSUE THREAD

**Issue by ckipp01** [lihaoyi/mill#989]
Given the following code snippet:
```scala
package example

object Hello extends Greeting with App {
  println(greeting)
}

trait Greeting {
  lazy val greeting: String = "hello"
  val x: Int = "hi" 
               ^^^^ // I'd expect the diagnostic to show this
}
```
According to the [BSP spec](https://build-server-protocol.github.io/docs/specification.html#publish-diagnostics):

> where Diagnostic is defined as it is in LSP.

And in the [LSP spec](https://microsoft.github.io/language-server-protocol/specification#range), the `Range` is zero based. However, it seems that the `Range` coming back from mill is not and the character also seems to be the character on the page rather than the line. For example, the correct diagnostic for the above should look like:

```
[Trace - 05:57:32 PM] Received notification 'build/publishDiagnostics'
Params: {
  "textDocument": {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/mill-tester/MillTester/src/example/Hello.scala"
  },
  "buildTarget": {
    "uri": "file:/Users/ckipp/Documents/scala-workspace/mill-tester/MillTester/?id\u003dMillTester"
  },
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 8,
          "character": 15
        },
        "end": {
          "line": 8,
          "character": 19
        }
      },
      "severity": 1,
      "code": "  val x: Int \u003d \"hi\"",
      "source": "bloop",
      "message": "type mismatch;\n found   : String(\"hi\")\n required: Int"
    }
  ],
  "reset": true
}
```
However, from Mill it looks like this:
```
[Trace - 06:08:28 PM] Received notification 'build/publishDiagnostics'
Params: {
  "textDocument": {
    "uri": "file:/Users/ckipp/Documents/scala-workspace/mill-tester/MillTester/src/example/Hello.scala"
  },
  "buildTarget": {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/mill-tester/MillTester/"
  },
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 9,
          "character": 151
        },
        "end": {
          "line": 9,
          "character": 151
        }
      },
      "severity": 1,
      "code": "  val x: Int \u003d \"hi\"",
      "source": "compiler from mill",
      "message": "type mismatch;\n found   : String(\"hi\")\n required: Int"
    }
  ],
  "reset": true
}
```
Notice the incorrect line and character.

**Comment by lefou** [comment]
Maybe trying out #999 and reporting if it helps or is still incorrect (e.g. off by one) would help.

**Comment by lefou** [comment]
Closed in #999
